client/network/src/schema/api.v1.proto

```diff
@@ -68,3 +68,23 @@ message BlockData {
 	bytes justifications = 8; // optional
 }
 
+// Request storage data from a peer.
+message StateRequest {
+	// Block header hash.
+	bytes block = 1;
+	// Start from this key.
+	bytes start = 2; // optional
+	// Include proof.
+	bool proof = 3;
+}
+
+// Response to `BlockRequest`
+message StateResponse {
+	// A collection of keys.
```

Comment:
For the case without proof, in #8431 https://github.com/paritytech/substrate/blob/844fb80e0195ed05a2ecc1d95baa0c5760278388/client/db/src/export_import.rs#L574 I write only key deltas, but it is not something that is  a must have since standard compression method would probably achieve similar size gain (but that is not very complex implementation).